src/libsyntax/tokenstream.rs

```diff
@@ -532,12 +685,14 @@ impl TokenSlice {
     }
 
     /// Returns the RHS of an assigment.
-    pub fn maybe_assignment(&self) -> Option<&TokenSlice> {
+    pub fn maybe_assignment(&self) -> Option<TokenStream> {
```

Comment:
I think this is a fine feature for parsing something like a `let loop` form as:
  
    let mut sum = 0;
    loop_fn!(loopfn (i = 0, j = 0)
        if i > 10 && j > 10 {
           sum
        } else if j > 10 {
           loopfn(i+1, 0)
        } else {
           sum += j;
           loopfn(i+1, 0)
        }
    )

In general, these operators are there to make macro writers' lives easier until `matches!` is implemented.